Calculator/Calculator/Model/FormManager.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  FormManager.swift
+//  Calculator
+//
+//  Created by mint, Whales on 2023/06/14.
+//
+
+import Foundation
+
+enum FormManager {
+    static let numberFormatter = NumberFormatter()
+    
+    static var configuredNumberFormatter : NumberFormatter {
+        self.numberFormatter.numberStyle = .decimal
+        self.numberFormatter.maximumFractionDigits = 19
+        self.numberFormatter.maximumIntegerDigits = 20
+        
+        return self.numberFormatter
+    }
+    
+    static func transformResult(from input: String) -> String {
+        var numberString = deleteComma(input)
+        numberString = deleteLastDot(numberString)
+        
+        guard let number = Double(numberString),
+              let output = self.configuredNumberFormatter.string(from: number as NSNumber) else {
+            return "NaN"
+        }
+        
+        guard output != "-0" else {
+            return "0"
+        }
+        
+        return output
+    }
+    
+    private static func deleteComma(_ input: String) -> String {
+        return input.replacingOccurrences(of: ",", with: "")
+    }
+    
+    private static func deleteLastDot(_ input: String) -> String {
+        guard input.hasSuffix(".") == false else {
+            return input.replacingOccurrences(of: ".", with: "")
+        }
+        
+        return input
```

Comment:
여기도 마찬가지로 조기 탈출보다는 분기의 느낌이 강하네요. if else문이 더 좋겠죠?
게다가 if else를 쓰면 `== false`가 빠지니까 오히려 코드읽기가 더 쉽네요! 😁
```suggestion
        if input.hasSuffix(".") {
            return input.replacingOccurrences(of: ".", with: "")
        } else {
            return input
        }
```

Reply:
gurad 문은 그대로 쓰지만 false는 제거하는 방법으로 바꿔 보았습니다..!

```swift
guard input.hasSuffix(".") else {
            return input
        }
        
        return input.replacingOccurrences(of: ".", with: "")
```